### PR TITLE
Add Login Page

### PR DESCRIPTION
**Pull request by KobbyMmo**
## Description

This creates a simple isolated login page

Fixes # [(181401117)](https://www.pivotaltracker.com/story/show/181401117)

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## Desktop Screenshots

image

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/pages/login.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      margin: "0 auto",
+      width: typography.pxToRem(widths.values.lg),
+    },
+  },
+  footer: {
+    marginTop: 0,
+  },
+}));
+
+function SignIn({ providers: providersProp, ...props }) {
```
Comment (kilemensi) [inline src/pages/login.js:28]:
```suggestion
function Login({ providers: providersProp, ...props }) {
```
Or why `/login` then `SignIn`?

File: src/pages/login.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+};
+
+SignIn.defaultProps = {
+  providers: undefined,
+};
+
+/**
+ *  NOTE: [getProviders()](https://next-auth.js.org/v3/getting-started/client#getproviders)
+ *                   calls /api/providers, which is not available during build
+ *                   time and hence we can't use it here.
```
Comment (kilemensi) [inline src/pages/login.js:56]: Something is missing here ... this is in relation to what?

**Review by kilemensi — COMMENTED** [review]
👍🏽 
----

Lets add a place to show errors redirected from [here](https://github.com/CodeForAfrica/PromiseTracker/blob/9b16bccebdc7ee5d5c4d3671844b7fe5b291301d/src/pages/api/auth/%5B...nextauth%5D.js#L50)

File: src/pages/login.js (hunk 1 above)
Comment (KobbyMmo) [reply src/pages/login.js:28]: > Lets add a place to show errors redirected from [here](https://github.com/CodeForAfrica/PromiseTracker/blob/9b16bccebdc7ee5d5c4d3671844b7fe5b291301d/src/pages/api/auth/%5B...nextauth%5D.js#L50)


Was thinking of a 500 page.
If we are reusing this page, then we may not need the 500 page?

**Comment by KobbyMmo** [comment]
> Lets add a place to show errors redirected from [here](https://github.com/CodeForAfrica/PromiseTracker/blob/9b16bccebdc7ee5d5c4d3671844b7fe5b291301d/src/pages/api/auth/%5B...nextauth%5D.js#L50)


Was thinking of a 500 page.
If we are reusing this page, then we may not need the 500 page? @kilemensi

**Comment by kilemensi** [comment]
> > Lets add a place to show errors redirected from [here](https://github.com/CodeForAfrica/PromiseTracker/blob/9b16bccebdc7ee5d5c4d3671844b7fe5b291301d/src/pages/api/auth/%5B...nextauth%5D.js#L50)
> 
> Was thinking of a 500 page. If we are reusing this page, then we may not need the 500 page? @kilemensi

Not necessarily @KobbyMmo ... what if we can't reach google sheet, WP backend, etc. We'll need to show 500 error page for all of those.

File: src/pages/login.js (hunk 2 above)
Comment (KobbyMmo) [reply src/pages/login.js:56]: i converted the page to serverside props, so not needed again

File: src/components/LoginPage/index.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+  useEffect(() => {
+    setErrorMessage(getErrorDescription(Router.query?.error));
+  }, []);
+
+  useEffect(() => {
+    if (session) {
+      Router.push("/act-now");
+    }
+  }, [session]);
```
Comment (kilemensi) [inline src/components/LoginPage/index.js:33]: Why are we checking `session` here and not in the `getServerSideProps` of `/login`?

File: src/utils/auth/error-list.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    case "EMAIL_REQUIRES_ADAPTER_ERROR":
+    case "CALLBACK_CREDENTIALS_JWT_ERROR":
+    case "CALLBACK_CREDENTIALS_HANDLER_ERROR":
+    case "PKCE_ERROR":
+    case "JWT_SESSION_ERROR":
+    case "SESSION_ERROR":
+    case "SIGNOUT_ERROR":
+      description =
+        "Authentication Failed. Please check your credentials and try again.";
+      break;
```
Comment (kilemensi) [inline src/utils/auth/error-list.js:26]:
```suggestion
  switch (code) {
    case "OAUTH_GET_ACCESS_TOKEN_ERROR":
    case "OAUTH_V1_GET_ACCESS_TOKEN_ERROR":
    case "OAUTH_GET_PROFILE_ERROR":
    case "OAUTH_PARSE_PROFILE_ERROR":
    case "GET_AUTHORIZATION_URL_ERROR":
    case "SIGNIN_OAUTH_ERROR":
    case "CALLBACK_OAUTH_ERROR":
    case "SIGNIN_EMAIL_ERROR":
    case "CALLBACK_EMAIL_ERROR":
    case "EMAIL_REQUIRES_ADAPTER_ERROR":
    case "CALLBACK_CREDENTIALS_JWT_ERROR":
    case "CALLBACK_CREDENTIALS_HANDLER_ERROR":
    case "PKCE_ERROR":
    case "JWT_SESSION_ERROR":
    case "SESSION_ERROR":
    case "SIGNOUT_ERROR":
      return "Authentication Failed. Please check your credentials and try again.";
```

File: src/utils/auth/error-list.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    default:
+      description = "An error occurred. Please try again.";
+      break;
+  }
+  return description;
```
Comment (kilemensi) [inline src/utils/auth/error-list.js:31]:
```suggestion
    default:
      return "An error occurred. Please try again.";
  }
```

File: src/components/LoginPage/index.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+  const providers = Object.values(providersProp ?? {});
```
Comment (kilemensi) [inline src/components/LoginPage/index.js:35]:
```suggestion
  const providers = providersProp ? Object.values(providersProp) : undefined;
```
Why waste {} (and the empty [] returned from Object.values)?

**Review by kilemensi — COMMENTED** [review]
👍🏽

**Review by kilemensi — COMMENTED** [review]
👍🏽 

-----

1. <https://promisetracker-git-feature-login-page-codeforafrica.vercel.app/login> is timing out,
2. /login doesn't follow PromiseTracker designs. From fonts to form not being centred on page, etc.
Screenshot from 2022-03-01 09-11-28

**Comment by KobbyMmo** [comment]
> * https://promisetracker-git-feature-login-page-codeforafrica.vercel.app/login is timing out,

I think the session is the culprit. I trying to find out more, but when i move it to the frontend, it works alright @kilemensi

File: src/pages/login.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+  return (
+    <Page
+      {...props}
+      classes={{ section: classes.section, footer: classes.footer }}
+    >
+      <LoginPage
+        classes={{ section: classes.section }}
+        providers={providersProp}
+        {...props}
```
Comment (kilemensi) [inline src/pages/login.js:54]: Only pass what `LoginPage` needs.

**Review by kilemensi — APPROVED** [review]
🚀 

------

For the buttons, use the respective sizes as on act-now page.

Screenshot from 2022-03-02 17-16-23

**Pull request merged**